kai-analyzer/pkg/service/analyzer.go

```diff
@@ -22,84 +22,91 @@ type Analyzer struct {
 	cancelFunc context.CancelFunc
 
 	initedProviders map[string]provider.InternalProviderClient
-	ruleSets        []engine.RuleSet
 }
 
-func NewAnalyzer(limitIncidents, limitCodeSnips, contextLines int, location, incidentSelector string, ruleFiles []string, log logr.Logger) (*Analyzer, error) {
+func NewAnalyzer(log logr.Logger) *Analyzer {
+	return &Analyzer{
+		Logger:          log,
+		initedProviders: map[string]provider.InternalProviderClient{},
+	}
+}
+
+func (a *Analyzer) Initialize(args InitializeArgs) error {
 	ctx, cancelFunc := context.WithCancel(context.Background())
 	eng := engine.CreateRuleEngine(ctx,
```

Comment:
Does this have to be create per initialized thing, or can it be shared?

Reply:
I wanted to make the server startup as dumb as possible, accepting all configuration items on the "initialize" request after the server has started. It was my understanding that, when you start gopls you would tell it things like "location" on the `initialize` method call. Maybe I need to research it more thoroughly.